notebook/nbconvert/handlers.py

```diff
@@ -116,13 +122,25 @@ def get(self, format, path):
             filename = os.path.splitext(name)[0] + resources['output_extension']
             self.set_header('Content-Disposition',
                                'attachment; filename="%s"' % escape.url_escape(filename))
-
         # MIME type
         if exporter.output_mimetype:
             self.set_header('Content-Type',
                             '%s; charset=utf-8' % exporter.output_mimetype)
+        self.finish(output) 
+
+    @web.authenticated
+    def get(self, format, path):
+        self.call_nbconvert(format, path, config=self.config)
+    
+    @web.authenticated
+    def post(self, format, path):
+        c = Config(self.config)
+        json_upload = self.get_json_body()
+        json_config = json.loads(json_upload.get("config",{}))
+        c.merge(json_config)
+        nb_content = json.dumps(json_upload["notebook"])
+        self.call_nbconvert(format, path, config=c, content=nb_content, post=True)
```

Comment:
It's actually not adding a new handler (from the perspective of the web endpoints), but merely distinguishing whether it got a GET or a POST request. I was trying to keep the code as similar in the conversion step as possible and that seemed the cleanest way to do it. 

I'm going to write something longer in the main PR comment thread based on a conversation with Matthias. But the approach I took is the way to have the cleanest backwards compatibility story. We're going to have to come up with a migration story for the more comprehensive version of this feature.